HabitRPG/habitica
# Screen reader labels for Inventory views

# what

Add a bunch of Aria-labels to various shop and equipment items
# why

User mentions that much of the site is unlabeled for screen-readers [on a Trello card mainly concerned](https://trello.com/c/pxTcbKuD/381-accessibility-settings) with the chat ordering and sure enough when using a screen readers several areas are impossible to navigate.

If your curious about how widely aria-label works as an option for screen-readers, check [out this handy article](https://www.nczonline.net/blog/2013/04/01/making-accessible-icon-buttons/) that does some testing with multiple browsers.
# known issues

While the labelling now uses a nice bit of Jquery to grab either the popover-title or popover text to make the aria label, that JS is added to the Options pages with the extremely hacky step of including a js file in its folder in `/views` (!!) 

I know this isn't the best approach in general but the site's pattern didn't present an obvious place to store this script otherwise. I'm not certain we want this labelling happening on other pages, so for now it's stored there.

In all cases the label is drawn from the popup title, and in a couple of instances there might be a more useful string to add as the label. Right now for example it doesn't read how many of an item you have, and I don't think the badge number will be picked up by any screen readers. When I tried grabbing the badge values... everything broke so I rolled that back.
# test

0) turn on a screen-reading program (on OSX you can do this with CMD+f5) and see that there are no labels on many equipment and inventory items, they just say things like 'potion button' and 'equipment button'
1) switch to branch
2) go to the inventory pages and see that nothing is messed up on these pages for a sighted user
3) turn on a screen-reading program (on OSX you can do this with CMD+f5) and see those same things have labels!
3a) if you aren't familiar with screen readers (it took me some time to get in control of the Apple version), just inspect the buttons and see they now have reasonable 'aria-label's attached.
